cmd/juju/application/updateseries.go

```diff
@@ -0,0 +1,182 @@
+
+See also:
+    machines
+    status
+    upgrade-charm
+`
+
+func (c *updateSeriesCommand) Info() *cmd.Info {
+	return &cmd.Info{
+		Name:    "update-series",
+		Args:    "[<application>|<machine>] <series>",
+		Purpose: "Update an application or machine's series.",
+		Doc:     updateSeriesDoc,
+	}
+}
+
+func (c *updateSeriesCommand) SetFlags(f *gnuflag.FlagSet) {
+	c.ModelCommandBase.SetFlags(f)
+	f.BoolVar(&c.force, "force", false, "Update even if the series is not supported by the charm and/or related subordinate charms.")
+}
+
+// Init implements cmd.Command.
+func (c *updateSeriesCommand) Init(args []string) error {
+	switch len(args) {
+	case 2:
+		if names.IsValidApplication(args[0]) {
+			c.applicationName = args[0]
+		} else if names.IsValidMachine(args[0]) {
+			c.machineNumber = args[0]
+		} else {
+			return errors.Errorf("invalid application name or machine number %q", args[0])
+		}
+		c.series = args[1]
+	case 1, 0:
+		return errors.Errorf("No application/machine and/or series specified")
+	default:
+		return cmd.CheckEmpty(args[2:])
+	}
+	return nil
+}
+
+// Run connects to the specified environment and starts the application or machine
+// update process.
+func (c *updateSeriesCommand) Run(ctx *cmd.Context) error {
+
+	if c.updateSeriesClient == nil && c.charmClient == nil {
+		apiRoot, err := c.NewAPIRoot()
+		if err != nil {
+			return errors.Trace(err)
+		}
+		c.apiRoot = apiRoot
+		c.updateSeriesClient = application.NewClient(c.apiRoot)
+		c.charmClient = charms.NewClient(c.apiRoot)
+	}
+
+	if c.apiRoot.BestFacadeVersion("Application") < 5 {
+		suffix := "this server"
+		if version, ok := c.apiRoot.ServerVersion(); ok {
+			suffix = fmt.Sprintf("server version %s", version)
+		}
+		return errors.New("updating the application series is not supported by " + suffix)
+	}
+
+	if c.machineNumber != "" {
+		//return c.updateMachineSeries(ctx)
+	}
+
+	return c.updateApplicationSeries(ctx)
+}
+
+func (c *updateSeriesCommand) updateApplicationSeries(ctx *cmd.Context) error {
+
+	applicationInfo, err := c.updateSeriesClient.Get(c.applicationName)
+	if err != nil {
+		return errors.Trace(err)
+	}
+
+	deployedSeries := applicationInfo.Series
+	if c.series == deployedSeries {
+		return errors.New(fmt.Sprintf("series %s is already in use by %s", c.series, c.applicationName))
+	}
+
+	url, err := c.updateSeriesClient.GetCharmURL(c.applicationName)
+	if err != nil {
+		return errors.Trace(err)
+	}
+
+	charmInfo, err := c.charmClient.CharmInfo(url.String())
+	if err != nil {
+		return errors.Trace(err)
+	}
+	supportedSeries := charmInfo.Meta.Series
+
+	_, seriesSupportedErr := charm.SeriesForCharm(c.series, supportedSeries)
+	if seriesSupportedErr != nil && !c.force {
+		return errors.New(fmt.Sprintf("series %s is not supported by the charm version in use. Use --force to override.", c.series))
+	}
+
+	return block.ProcessBlockedError(
+		c.updateSeriesClient.UpdateApplicationSeries(c.applicationName, c.series, c.force),
```

Comment:
We are not processing the errors from the server side and we should be. 
eg the api call result maybe unsupported series error and so the relevant message needs to be printed.

We could make the argument that the API call above to check the series is not needed. Perform validation in the CLI if it's cheap (ie not requiring an API call).